docs/howto/custom-rootfs-multiple-targets.md

```diff
@@ -0,0 +1,314 @@
+# How-to deploy a custom rootfs across multiple Windows machines with Landscape API
+
+This guide shows how to use the Landscape API to deploy a custom rootfs
+with some cloud-config user data for further initialisation of the distro
+instances via cloud-init across multiple Windows machines already registered
+with Landscape via Ubuntu Pro for WSL.
+It can be followed either on Windows (using PowerShell) or Linux (using bash).
+
+## Prerequisites
+
+- A running self-hosted Landscape server version `24.10~beta.5` or later;
+- Multiple Windows machines already registered with Landscape via Ubuntu Pro for WSL;
+- Make sure you have installed `curl` and `jq`, if you're following this guide on a Linux machine.
+- Familiarity with Bash and/or PowerShell.
+
+## Prepare the environment
+
+To ease the subsequent commands, adjust and export the following environment variables:
+
+`````{tabs}
+
+````{group-tab} Bash
+```bash
+# Credentials to authenticate the API requests
+export LANDSCAPE_USER_EMAIL=admin@mib.com
+export LANDSCAPE_USER_PASSWORD=mib
+export LANDSCAPE_URL=https://landscape.mib.com
+
+# The URL of the custom rootfs to be deployed:
+export ROOTFS_URL="http://landscape.mib.com:9009/ubuntu-24.04-custom.tar.gz"
+
+# The list of IDs of the different Windows machines on which we are going to deploy WSL instances:
+export PARENT_COMPUTER_IDS=(26 30 31)
+
+# The name of the WSL instance to be created:
+export COMPUTER_NAME=Carbonizer
+
+# Path to the cloud-config file whose contents will be used to initialize the WSL instances:
+export CLOUD_INIT_FILE="~/Downloads/init.yaml"
+```
+
+````
+
+````{group-tab} PowerShell
+```powershell
+# Credentials to authenticate the API requests
+$LANDSCAPE_USER_EMAIL="admin@mib.com"
+$LANDSCAPE_USER_PASSWORD="mib"
+$LANDSCAPE_URL="https://landscape.mib.com"
+
+# The URL of the custom rootfs to be deployed:
+$ROOTFS_URL="http://landscape.mib.com:9009/ubuntu-24.04-custom.tar.gz"
+
+# The list of IDs of the different Windows machines on which we are going to deploy WSL instances:
+$PARENT_COMPUTER_IDS=@(26, 30, 31)
+
+# The name of the WSL instance to be created:
+$COMPUTER_NAME="Carbonizer"
+
+# Path to the cloud-config file whose contents will be used to initialize the WSL instances:
+$CLOUD_INIT_FILE="~/Downloads/init.yaml"
```

Comment:
Just checking: is the unix-style path used here OK in the PS script?

Reply:
Yes, it it's fine. But it won't hurt to make it into Windows paths.

Reply:
Yes, I was just thinking about someone reading this from a Windows-specific background.
In that sense, I was also wondering if the "~" is OK in the path but it's probably OK.

Reply:
Yeah, that's OK on Powershell. Not so OK for the old cmd.exe, but we don't want to attempt to write an equivalent batch script for that thing :)

Reply:
It could have been the third tab Carlos!!! :laughing: